iwyu_ast_util.cc

```diff
@@ -155,6 +155,14 @@ void DumpASTNode(llvm::raw_ostream& ostream, const ASTNode* node) {
   }
 }
 
+clang::TemplateSpecializationKind GetTemplateSpecializationKind(
+    const Decl* decl) {
+  if (const clang::CXXRecordDecl* record = DynCastFrom(decl)) {
```

Comment:
See below -- I think we should prefer `const auto* x = dyn_cast<T>(y)` over `const T* x = DynCastFrom(y)`.

Reply:
Alright. I actually thought we settled for `DynCasFrom`, but yeah, I also think `dyn_cast` is a better choice.